### ISSUE TITLE
Question about `scene_extent`

### ISSUE THREAD

**Issue by XuyangBai** [graphdeco-inria/gaussian-splatting#48]
Why is the scene_extent computed as the radius of camera center rather than for example something related to the initial pointcloud? In your implement, the cameras with the same trajectories but one set of camera facing inward and one set of cameras facing outward will have exactly the same scene_extent, but the former one may corresponds with a smaller scene. 

Also the resulting scene_extent is used to scale up the position learning rate. I still quite not understand why the learning rate should be proportional to the radius of camera center.

**Comment by grgkopanas** [comment]
> Why is the scene_extent computed as the radius of camera center rather than for example something related to the initial pointcloud?
The initial point-cloud often suffers from outliers in realistic scenarios, where cameras are usually more well behaved. Of course it's not perfect it's rather a quanitity that is relative and reasons about the size of the scene.

> Also the resulting scene_extent is used to scale up the position learning rate:
This was more of an experimental observation that we realised that some times it helped. The reason why its helpful has to do with the fact that we use Adam as an optimizer. If you go through the formulation of Adam you will see that in the simplest case where a point gets a constant gradient all the time it will move with a constant step in world-space no matter the size of the scene.
This is not a bullet-proof solution but it works well for the datasets that we tried. We encourage the users to try to experiment with the learning rates and other potential hyper-parameters.

Best,
George

**Comment by XuyangBai** [comment]
Hi George, Thanks a lot for the explanation. 

> If you go through the formulation of Adam you will see that in the simplest case where a point gets a constant gradient all the time it will move with a constant step in world-space no matter the size of the scene.

And I also found some discussion in other issue
> there's two levels of spatial discrepancies:
> - gradients have different sizes due to projection (enlarge the same scene -> smaller gradients in world space)
> - learning rates need to adapt to fit enlarged scenes (larger scene -> actually need to move more)


So based on the above analyzation, it's expected to use a larger positional learning rate for a larger scene, that's why you use scene extent to scale up the positional learning rate. But it might not be a linear relationship, so you lower the position_lr_init by 0.1 in the readme for the driving scene data, is my understand correct?

**Comment by Snosixtyboo** [comment]
Hi,

we need to distinguish "large" as in the coordinates it covers and the scene complexity. 

For the first: assume a scene with a single object (e.g., a truck or train), with cameras whose coordinates range from -1 to 1 in XYZ and a point cloud in -2 to 2. But it would be just as valid to have camera coordinates from -100 to 100 and a point cloud from -200 to 200: the concrete values depend on your calibration method! Scaling up learning rates by the camera extent is supposed to make the single-object scene work the same, regardless of your calibration's coordinate range.

For the second: Our learning rates are set by default to target a central observed object or a couple of objects. But if you are looking at **hundreds** of them, e.g., in an urban dataset, then each object occupies only a tiny portion of the scene. In that case, our learning rates are too high: With a single update, Gaussians may move by more than an entire object's length and as a result, the scene diverges. Therefore, it would have to be scaled down manually.

Hth,
Bernhard

**Comment by XuyangBai** [comment]
Thanks a lot for the explanation! @grgkopanas @Snosixtyboo
